### PR TITLE
[enhancement](inverted index) Support fulltext index evaluate equal query and list query

### PR DESCRIPTION
**Pull request by Tanya-W**
# Proposed changes

Issue Number: close #xxx

## Problem summary

fulltext index is  the inverted index of the specified tokenizer, before this pr, fulltext index only can evaluate match predicate, this pr to support evaluate equal predicate and list predicate.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -550,21 +550,35 @@ bool SegmentIterator::_check_apply_by_bitmap_index(ColumnPredicate* pred) {
...
+    bool handle_by_fulltext = _column_has_fulltext_index(unique_id);
+    if (handle_by_fulltext) {
+        // when predicate in compound condition which except leafNode of andNode,
+        // only can apply match query for fulltext index,
+        // when predicate is leafNode of andNode,
+        // can apply 'match qeury' and 'equal query' and 'list query' for fulltext index.
+        return (pred_in_compound ? pred->type() == PredicateType::MATCH
+                                 : (pred->type() == PredicateType::MATCH ||
+                                    PredicateTypeTraits::is_eqaul_and_list(pred->type())));
+    }
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:580]: Is it right to return true if handle_by_fulltext == false ?

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 2]
```diff
@@ -679,30 +698,21 @@ bool SegmentIterator::_is_handle_predicate_by_fulltext(int32_t unique_id) {
...
+                                             PredicateTypeTraits::is_eqaul_and_list(pred->type());
         roaring::Roaring bitmap = _row_bitmap;
         Status res =
                 pred->evaluate(_schema, _inverted_index_iterators[unique_id], num_rows(), &bitmap);
         if (!res.ok()) {
             if ((res.code() == ErrorCode::INVERTED_INDEX_FILE_NOT_FOUND &&
                  pred->type() != PredicateType::MATCH) ||
-                res.code() == ErrorCode::INVERTED_INDEX_FILE_HIT_LIMIT) {
+                res.code() == ErrorCode::INVERTED_INDEX_FILE_HIT_LIMIT ||
+                (res.code() == ErrorCode::INVERTED_INDEX_NO_TERMS &&
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:714]: add comment for NO_TERMS condition

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 1 above)
Comment (Tanya-W) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:580]: it is right, before check `handle_by_fulltext ` already check other case, if `handle_by_fulltext == false ` return true means can apply by bkd index reader or string type index reader

**Comment by Tanya-W** [comment]
run buildall

**Comment by Tanya-W** [comment]
run compile

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**